[Instrumentation.AWSLambda] Enable nullable (#1295)
open-telemetry/opentelemetry-dotnet-contrib: src/OpenTelemetry.Instrumentation.AWSLambda/CHANGELOG.md

## Unreleased

* BREAKING: `ILambdaContext context` argument of all tracing methods of

>   `OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper` became nullable.

Reviewer: ```suggestion
  `OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper` was annotated as non-nullable.
```

> * Enabled nullable for `OpenTelemetry.Instrumentation.AWSLambda` and `OpenTelemetry.Instrumentation.AWSLambda.Tests`

Reviewer: ```suggestion
* Enabled null state analysis for `OpenTelemetry.Instrumentation.AWSLambda` and `OpenTelemetry.Instrumentation.AWSLambda.Tests`. The interface will now contain attributes for null-state static analysis. If null state analysis is enabled your depending project, you may encounter new warnings.
```

## Unreleased

* BREAKING: `ILambdaContext context` argument of all tracing methods of
  `OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper` was annotated as non-nullable.
* Enabled null state analysis for `OpenTelemetry.Instrumentation.AWSLambda`.
  The interface will now contain attributes for null-state static analysis.

>   If null state analysis is enabled your depending project, you may encounter new warnings.

Author: Fixed.

## Unreleased

* BREAKING: `ILambdaContext context` argument of all tracing methods of
  `OpenTelemetry.Instrumentation.AWSLambda.AWSLambdaWrapper` was annotated as non-nullable.
* Enabled null state analysis for `OpenTelemetry.Instrumentation.AWSLambda` and

>   `OpenTelemetry.Instrumentation.AWSLambda.Tests`. The interface will now

Author: Removed mentioning of the test project.

  contain attributes for null-state static analysis. If null state analysis is
  enabled your depending project, you may encounter new warnings.

>   ([#894](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/894))

Reviewer 2: It should be a link to PR not a general issue.

```suggestion
  ([#1295](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1295))
```

Author: Then it should be:
([#1295](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1295))

Reviewer 2: suggestion fixed
